### ISSUE TITLE
[Feature]: Use reference to System.Net.Http using framework `Reference` instead of PackageReference 

### ISSUE THREAD

**Issue by jacekmlynek** [fluentassertions/fluentassertions#2120]
### Background and motivation

I hope this is the correct place to raise this.

Currently for .net47 target you are adding reference to `System.Net.Http` as package dependency: 

```
 <ItemGroup Condition="'$(TargetFramework)' == 'net47'">
    <PackageReference Include="System.Net.Http" Version="4.3.4" />
  </ItemGroup>
```
According to this [issue](https://github.com/dotnet/runtime/issues/26131) dot net team advise adding `System.Net.Http` standard framework reference instead. Just quoting them:

> if you're building a .NET Framework 4.7.2 class library, then you could just add a normal Framework reference to System.Net.Http.
> 
> In most cases, we don't advise people use the separate System.Net.Http NuGet package anymore. See https://github.com/dotnet/runtime/issues/18280 and https://github.com/dotnet/runtime/issues/20777.

The main reason why I am suggesting this change is that VisualStudio resolve 4.7.2 project dependency to `System.Net.Http` and all its System dependencies by adding references to .net framework installed on local machine instead of project nuget. I think this issue has been raise in multiple place e.g. [Broken System.Net.Http 4.1.1-4.3.0 post-mortem](https://github.com/dotnet/runtime/issues/20777#issuecomment-338418610) . It can also be problematic when you have reference to same assembly with both a single name reference and a package reference. For more details please take a look on [Single-name references are removed by the SDK when targeting .NET Framework 4.7.2 and referencing a package with the same assembly](https://github.com/Microsoft/dotnet/blob/main/releases/net472/KnownIssues/613745%20-%20Single-name%20references%20are%20removed%20by%20the%20SDK%20when%20targeting%204.7.2.md)

### Alternative Concerns

Could you consider changing above to something like:

```
<ItemGroup Condition=" '$(TargetFramework)' == 'net47' ">
    <Reference Include="System.Net.Http" />
</ItemGroup>
```

It looks like another few other projects decide to follow this advise e.g. [IdentityModel](https://github.com/IdentityModel/IdentityModel/issues/76), [Microsoft Graph](https://github.com/microsoftgraph/msgraph-sdk-dotnet/issues/312)

**Comment by eNeRGy164** [comment]
Is this related to https://github.com/fluentassertions/fluentassertions/pull/1806?

**Comment by jacekmlynek** [comment]
@eNeRGy164 if you ask me, I can say that changes from PR will solve my problem, but #1805, was more about linux runtime and framework 4.8. Definitely final solution mention in last comment will not works for me.

**Comment by jnyrup** [comment]
Just to get a more complete understanding.
Is this causing a real problem for you or are you pointing out the pitfalls?

**Comment by jacekmlynek** [comment]
@jnyrup yes that cause real problem for some of my projects that based on 4.7.2.

**Comment by jnyrup** [comment]
Alright 👍 
Can you provide a complete example which reproduces a problem?

**Comment by dennisdoomen** [comment]
Just read through all the references, and it seems to be a valid problem. Even Claire Novotny was pulled in to help the IdentityModel folks understand this. My only concern is how this affects existing code bases that would upgrade.

**Comment by dennisdoomen** [comment]
Is this a coincidence? https://github.com/fluentassertions/fluentassertions/issues/2121

**Comment by dennisdoomen** [comment]
I _am_ wondering why some of the .NET 4.7.2 projects I've been involved with didn't have that problem.

**Comment by dennisdoomen** [comment]
Could you try [this NuGet package](https://github.com/fluentassertions/fluentassertions/suites/10777068487/artifacts/541549779) (it's wrapped in an `artifacts.zip`)?

**Comment by dennisdoomen** [comment]
Instead, @AArnott produced an even [better PR](https://github.com/fluentassertions/fluentassertions/pull/2122). So can you try [this NuGet package](https://github.com/fluentassertions/fluentassertions/suites/10773776135/artifacts/542173591)?

**Comment by AArnott** [comment]
@dennisdoomen This _is_ an amazing coincidence.

**Comment by AArnott** [comment]
> Even Claire Novotny was pulled in to help the IdentityModel folks understand this.

I had to pull in one of Microsoft's MSBuild team engineers to understand it. :) (and I used to be on the MSBuild team myself!)

**Comment by jacekmlynek** [comment]
sorry, I wasn't able to answer during the weekend.

@dennisdoomen @AArnott I have take a look at #2122 - looks sensible to me and what is more import, the attached pre-release version solved my issues.

To be honest I do not fully understand why msbuild resolved `System.Net.Http` reference in this way and why in some projects behaviour is different The only thing that I was able to find suggest that this may be related to[ bindings in config](https://stackoverflow.com/questions/57803285/use-latest-version-of-system-net-http-in-net-framework)

I agree that insane coincidence :)

**Comment by IT-VBFK** [comment]
Fixed with [97cf195](https://github.com/fluentassertions/fluentassertions/commit/97cf1955b8f9d4fab6e4415353ea3599ee450c3e)?
